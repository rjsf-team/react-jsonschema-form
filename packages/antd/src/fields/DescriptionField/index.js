import React from 'react';

const DescriptionField = ({
  // autofocus,
  description,
  // disabled,
  // errorSchema,
  // formContext,
  // formData,
  id,
  // idSchema,
  // name,
  // onChange,
  // readonly,
  // registry,
  // required,
  // schema,
  // uiSchema,
}) => <span id={id}>{description}</span>;

export default DescriptionField;
