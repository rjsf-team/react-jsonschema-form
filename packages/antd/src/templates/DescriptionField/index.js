import React from "react";

const DescriptionField = ({
  description,
  id,
  // registry,
}) => <span id={id}>{description}</span>;

export default DescriptionField;
