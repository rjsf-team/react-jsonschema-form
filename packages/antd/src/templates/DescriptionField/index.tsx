import React from "react";
import { DescriptionFieldProps } from "@rjsf/utils";

const DescriptionField = ({ description, id }: DescriptionFieldProps) => {
  if (!description) {
    return null;
  }
  return <span id={id}>{description}</span>;
};

export default DescriptionField;
