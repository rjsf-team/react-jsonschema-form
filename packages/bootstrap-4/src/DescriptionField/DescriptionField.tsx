import React from "react";
import { FieldProps } from "@rjsf/core";

export interface DescriptionFieldProps extends Partial<FieldProps> {
  description?: string;
}

const DescriptionField = ({ description }: Partial<FieldProps>) => {
  if (description) {
    return <div><div className="mb-3">{description}</div></div>;
  }

  return null;
};

export default DescriptionField;
