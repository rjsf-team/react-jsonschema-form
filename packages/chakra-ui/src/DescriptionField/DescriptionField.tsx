import { DescriptionFieldProps, FormContextType, RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';
import { Text } from '@chakra-ui/react';

export default function DescriptionField<
  T = any,
  S extends StrictRJSFSchema = RJSFSchema,
  F extends FormContextType = any
>({ description, id }: DescriptionFieldProps<T, S, F>) {
  if (!description) {
    return null;
  }

  if (typeof description === 'string') {
    return (
      <Text as='sup' fontSize='md' id={id}>
        {description}
      </Text>
    );
  }

  return <>{description}</>;
}
