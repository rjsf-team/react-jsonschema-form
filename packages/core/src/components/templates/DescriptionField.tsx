import { DescriptionFieldProps, FormContextType, RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';

/** The `DescriptionField` is the template to use to render the description of a field
 *
 * @param props - The `DescriptionFieldProps` for this component
 */
export default function DescriptionField<
  T = any,
  S extends StrictRJSFSchema = RJSFSchema,
  F extends FormContextType = any
>(props: DescriptionFieldProps<T, S, F>) {
  const { id, description } = props;
  if (!description) {
    return null;
  }
  if (typeof description === 'string') {
    return (
      <p id={id} className='field-description'>
        {description}
      </p>
    );
  } else {
    return (
      <div id={id} className='field-description'>
        {description}
      </div>
    );
  }
}
