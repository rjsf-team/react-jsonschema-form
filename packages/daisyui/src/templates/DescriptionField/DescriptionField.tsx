import { DescriptionFieldProps, StrictRJSFSchema, RJSFSchema, FormContextType } from '@rjsf/utils';

/** The `DescriptionField` component renders descriptive text for a form field
 * with DaisyUI styling. It displays the description in a subtle text color
 * with proper spacing.
 *
 * @param props - The `DescriptionFieldProps` for the component
 */
export default function DescriptionField<
  T = any,
  S extends StrictRJSFSchema = RJSFSchema,
  F extends FormContextType = any
>(props: DescriptionFieldProps<T, S, F>) {
  const { id, description } = props;
  return (
    <div id={id} className='description-field my-4'>
      <div className='text-sm text-base-content/80'>{description}</div>
    </div>
  );
}
