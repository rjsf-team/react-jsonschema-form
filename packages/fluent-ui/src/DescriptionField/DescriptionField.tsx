import React from "react";

import { FieldProps } from "@rjsf/core";

import { Text } from "@fluentui/react";

const DescriptionField = ({ description }: FieldProps) => {
  if (description) {
    return <Text>{description}</Text>;
  }

  return null;
};

export default DescriptionField;
