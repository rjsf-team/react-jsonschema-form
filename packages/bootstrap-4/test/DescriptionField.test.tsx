import React from "react";
import DescriptionField from "../src/DescriptionField";
import renderer from "react-test-renderer";

describe("DescriptionField", () => {
  test("should return null when no description as a props is passed", () => {
    const tree = renderer.create(<DescriptionField />).toJSON();
    expect(tree).toBe(null);
  });

  test("should return h2 element when description is being passed as props", () => {
    const tree = renderer
      .create(<DescriptionField description="SOME THING" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
