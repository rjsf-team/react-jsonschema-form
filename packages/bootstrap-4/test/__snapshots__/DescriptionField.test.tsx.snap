// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionField should return h2 element when description is being passed as props 1`] = `
<div>
  <div
    className="mb-3"
  >
    SOME THING
  </div>
</div>
`;
